### ISSUE TITLE
element or contained text wrap to nest line

### ISSUE THREAD

**Issue by imahdio** [imahdio/CSS-Essential-Training#8]
I'm studying Inline, block, and display lesson from [here](http://168.119.85.43/css-essential-training-3/04-The%20Box%20Model/)
![image](https://user-images.githubusercontent.com/64577273/148638685-bc31091e-324d-49e6-b641-59b146386594.png)

based above highlighted phrase , which concept be undrstood?
1. inline elements themeselves wrap to next line or
2. their content be wrapped
if the content can't fit on the same line.

relates image to:
state1|state2
-|-
![image](https://user-images.githubusercontent.com/64577273/148639129-40361300-91c8-45ee-964a-2245a66490c8.png)|![image](https://user-images.githubusercontent.com/64577273/148639144-734c4f69-0222-43b2-bd4e-ad092624eece.png)

**Comment by dprincecoder** [comment]
@imahdio 
Inline element will wrap to the next line if the content they consume is way longer than the container they fill.
So condition 1 will be applied.

**Comment by MBEDOBE** [comment]
what that means is that, when using an inline element, it does not break the flow of the text (it continues in the same line until the text can't fit the width of the container then it moves to the next line) unlike block-level elements which will break where the element starts from. there are elements which are referred to as inline and others block-level [https://developer.mozilla.org/en-US/docs/Web/HTML/Inline_elements#inline_vs._block-level_elements_a_demonstration](https://developer.mozilla.org/en-US/docs/Web/HTML/Inline_elements#inline_vs._block-level_elements_a_demonstration).
![block-level](https://user-images.githubusercontent.com/25242976/148643493-1be35a6b-b78e-4ba5-8396-4ef80acaf407.PNG)
![inline](https://user-images.githubusercontent.com/25242976/148643495-39be4b72-6546-4162-97a5-dcd4b77da5d5.PNG)
.

**Comment by imahdio** [comment]
tnx dear @MBEDOBE for your kindly reply
I'm agree with your opinion
but if aim of instructor is what you explained , she mistakenly used style1 image instead of style 2.

I mean the demonstarton image of this concept is only style 2 while the instror how the style 1 figure.
do you agree?

**Comment by MBEDOBE** [comment]
> tnx dear @MBEDOBE for your kindly reply I'm agree with your opinion but even if aim of instructor is what you explained , she mistakenly used style1 image instead of style 2.
> 
> I mean the demonstarton image of inline element is only style 2 while the instructor show the style 1 figure. do you agree?

for me image 1 is correct. take a look at this example i just made. (note: button is an inline element)
![inline2](https://user-images.githubusercontent.com/25242976/148643940-f3848fcb-956e-4615-9968-d5a77d010383.PNG)

**Comment by DevanshThakkar** [comment]
![Inline and Block](https://user-images.githubusercontent.com/88326907/148644156-3816d544-544a-4977-8991-264bd7183b9c.PNG)

**Comment by DevanshThakkar** [comment]
The major difference is that inline-block allows to set a width and height on the element. Also, with display: inline, top and bottom margins & paddings are not respected, and with display: inline-block they are.

**Comment by DevanshThakkar** [comment]
![Inline and Block2](https://user-images.githubusercontent.com/88326907/148644406-901d20cf-45b2-4750-93bd-fed4a05bf809.PNG)

**Comment by MBEDOBE** [comment]
> tnx for both @DevanshThakkar and @MBEDOBE for their cooperation But `<button>` is an inline-block element😔😕

that's true, from HTML4 `button` is now regarded as inline-block by most browsers. you can read Oriol's answer here to understand better [https://stackoverflow.com/questions/27764122/button-is-an-inline-element-but-width-is-working](https://stackoverflow.com/questions/27764122/button-is-an-inline-element-but-width-is-working)

**Comment by imahdio** [comment]
after all , do all agree to say such this feature:
he element's content will only wrap to the next line if the content can't fit on the same line within its container.

instead of 
The elements will only wrap to the next line if the content can't fit on the same line within its container.

and introduce this related image
![](https://user-images.githubusercontent.com/64577273/148639144-734c4f69-0222-43b2-bd4e-ad092624eece.png)

instead of
![](https://user-images.githubusercontent.com/64577273/148639129-40361300-91c8-45ee-964a-2245a66490c8.png)

**Comment by DevanshThakkar** [comment]
The inline element will let any other content overlap over its content, while the inline:block element will not let any other content be overlapped. It blocks the self generated space. (This is just FYI).
Thank You.

**Comment by MBEDOBE** [comment]
@imahdio  it will just move the element to the next line if the content is more than available space

![inline3](https://user-images.githubusercontent.com/25242976/148644835-7d8a1463-b4d6-411f-8e0f-c85bbdbb27bf.PNG)

**Comment by imahdio** [comment]
> @imahdio it will just move the element to the next line if the content is more than available space
> 
> ![inline3](https://user-images.githubusercontent.com/25242976/148644835-7d8a1463-b4d6-411f-8e0f-c85bbdbb27bf.PNG)

yes mate , it's because that's inline element
you can change its display
experiment with it in here
https://codepen.io/ma400/pen/YzrOPPV

**Comment by MBEDOBE** [comment]
> after all , do all agree to say such this feature for inline elements: the element's content will only wrap to the next line if the content can't fit on the same line within its container.
> 
> instead of
> 
> > The elements will only wrap to the next line if the content can't fit on the same line within its container.
> 
> **NOTICE: in my point of view , container in here must refer to viewport**
> 
> and demonstrate the inline elements behaviours by using this related image ![](https://user-images.githubusercontent.com/64577273/148639144-734c4f69-0222-43b2-bd4e-ad092624eece.png)
> 
> instead of ![](https://user-images.githubusercontent.com/64577273/148639129-40361300-91c8-45ee-964a-2245a66490c8.png)

you're right on this if you're referring to the viewport

**Comment by vidhu001** [comment]
@imahdio as my opinion an inline element does not start on a new line and only takes up as much width as necessary.

An inline-block  is an element is a unit that always remains a block (but within the text flow), which can only be moved as a whole and not be divided.

You can also change the visual presentation of an element using the CSS display property. For example, by changing the value of display from "inline" to "block", you can tell the browser to render the inline element in a block box rather than an inline box, and vice versa. However, doing this will not change the category and the content model of the element. For example, even if the display of the span element is changed to "block", it still would not allow to nest a div element inside it.
Here you can see main difference between in inline block elements and block elements :

Content model:Generally, inline elements may contain only data and other inline elements. You can't put block elements inside inline elements.

Formatting:By default, inline elements do not force a new line to begin in the document flow. Block elements, on the other hand, typically cause a line break to occur (although, as usual, this can be changed using CSS).

As you can see below the word wrap property as well:-

![Screenshot (139)](https://user-images.githubusercontent.com/71186753/148651518-63d151e3-b632-4a2b-b2b9-ad060993f7d1.png)

Thank you!
Regards 
Hope it is helpful!
